### PR TITLE
Revamp index

### LINKED ISSUE #116: Add way to visualize "needs longer ride"
Right now the mobility device management interface will show you which kind of mobility device needs a longer ride (it's an attribute on the object), but there's no way to easily see that from the passenger index.
It was requested that there be a way to see at-a-glance, specifically with styling and _not_ with another column, which passengers need more than a single ride slot.

### LINKED ISSUE #137: automatically check the 'active' box when editing any passenger

### LINKED ISSUE #139: rename button "print filtered list" instead of "print these filters"

### LINKED ISSUE #140: Re-do colors/labels
The colors and lables (will expire in one week, no-note, etc) for the passengers to distinguish between different major "types" of passenger are a little bit confusing, say the STSPVs and ST manager. I have a list of desired colors and labels that should make it more self-explanatory:

- [x] Normal remins white/clear
- [x] "Expires within one week" gets that label and remains yellow
- [x] Doctor's note expired but within the grace period / no note but within grace period is functionally the same, so combine those two states into one label "Within grace period, needs doctor's note". The color should be orange.
- [x] No doctor's note that's not expired (out of grace period) should have the label "expired - out of grace period"

### LINKED ISSUE #143: Remove email column from index
There's no need for the email to be on the index page--when emailing a passenger you can go to the show page.

### LINKED ISSUE #145: Add "needs longer ride" column
We'll be removing the email column #143 so it should make room. Just a checkmark, no X, next to mobility device.

### PR DESCRIPTION
**Pull request by Anbranin**
closes #145 
closes #143 
closes #140 
closes #139 
closes #137
closes #116 

These are all pretty tiny issues so I thought I'd wrap them up together. 
I'm meeting with ST today to discuss the colors, and also I need PR #134 to be closed first. But, I thought I'd let people look at this in the meantime if they wanted.

### CODE REVIEW

**Comment by werebus** [comment]
I'm a little confused by this PR. It looks to be a number of things I feel like I've already reviewed.

**Comment by Anbranin** [comment]
> It looks to be a number of things I feel like I've already reviewed.

This is what happens when you have too many PRs open. There _are_ some new things in this branch; I'm going to look into those items which are mysteriously duplicated.

**Comment by Anbranin** [comment]
Oh, @werebus I remember now, I'm sorry--you did review the fixed-header stuff, but that PR wasn't made off of master. It was made off of this branch. At the time I thought you were aware but should have made it clear.

**Comment by Anbranin** [comment]
```
The command "bundle exec codeclimate-test-reporter" exited with 0.
```
-_-

**Comment by werebus** [comment]
Coverage dropped.  On the plus side that coverage number is baloney (#157).

File: app/views/passengers/_form.html.haml [hunk 1]
```diff
@@ -46,7 +46,7 @@
       :id, :name), include_blank: 'None'
   .field
     = form.label :active?
-    = form.check_box :active?, {}, 'active', 'archived'
+    = form.check_box :active?, {checked: 'checked'}, 'active', 'archived'
```
Comment (werebus) [inline app/views/passengers/_form.html.haml]: I have a slight preference for `{checked: true}` for readability.

File: app/pdfs/passengers_pdf.rb [hunk 2]
```diff
@@ -29,11 +36,15 @@ def passengers_table(passengers)
 
   def passenger_row(passenger)
     name = passenger.name
+    needs_longer_ride = if passenger.mobility_device&.needs_longer_rides?
+                          "\u2714"
```
[the remark this answers is not in the document]
Comment (werebus) [reply app/pdfs/passengers_pdf.rb]: That is, I had to Google the Unicode codepoint (although I suspected it was a check-mark).

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -13,4 +13,9 @@ def yes_no_image(value)
     icon_class = value ? 'glyphicon-ok' : 'glyphicon-remove'
     content_tag :span, nil, class: "glyphicon #{icon_class}"
   end
+
+  def yes_image(value)
```
[the remark this answers is not in the document]
Comment (Anbranin) [reply app/helpers/application_helper.rb]: that's nice, I wouldn't have thought of using kwargs to clean it up.

File: app/pdfs/passengers_pdf.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (Anbranin) [reply app/pdfs/passengers_pdf.rb]: ummm duh everyone knows what the unicode for checkmark is 🙄

File: app/views/passengers/index.haml [hunk 4]
```diff
@@ -3,44 +3,45 @@
...
+        - if @current_user.admin?
           %th{data: {orderable: 'false'}}
           %th{data: {orderable: 'false'}}
     %tbody
       - @passengers.each do |passenger|
         %tr{class: passengers_table_row_class(passenger)}
           %td= passenger.name
           %td= passenger.mobility_device.try(:name)
-          %td= passenger.email
+          %td{data: {sort: passenger.mobility_device.try(:needs_longer_rides?).to_s}}= yes_image passenger.mobility_device.try(:needs_longer_rides?)
```
[the remark this answers is not in the document]
Comment (Anbranin) [reply app/views/passengers/index.haml]: Is the `present` there to coerce nil to false?

[the remark this answers is not in the document]
Comment (werebus) [reply app/views/passengers/index.haml]: Yeah. I have this sense that predicate methods should return strictly `true` and `false` (even though `nil` is pretty much also false).

Could also do `!!mobility_device&.needs_longer_rides`

[the remark this answers is not in the document]
Comment (Anbranin) [reply app/views/passengers/index.haml]: I was just making sure. I think the predicate method used further down should probably be changed but I've done a lot here.

**Comment by Anbranin** [comment]
I had some logic problems that I just fixed; people weren't getting colored the right way so I noticed it.
Let me know if any of it is unclear.

**Comment by Anbranin** [comment]
I TESTED IT AND EVERYTHING

File: app/models/passenger.rb [hunk 5]
```diff
@@ -33,6 +33,25 @@ def expiration_display
...
+    recently_registered = registration_date >= 3.business_days.ago
+    doctors_note&.expired_within_grace_period? ||
+    (doctors_note.blank? && recently_registered)
+  end
+
+  def rides_expired?
+    return false if permanent?
+
+    registration_expired = registration_date < DoctorsNote.grace_period
+    registration_expired && (doctors_note.nil? || doctors_note&.expired?)
```
Comment (werebus) [inline app/models/passenger.rb]: You don't need the `&.` here.  The short-circuiting of `||` means that `.expired?` is only called if `doctors_notes` is definitely not `nil?`

Comment (Anbranin) [reply app/models/passenger.rb]: Oh, excellent. Logic is hard.

**Comment by Anbranin** [comment]
<3 <3 <3

**Pull request merged**